File: frontend/src/features/orders/types/Order.ts
export type OrderStatus =
  | "Pending"
  | "Preparing"
  | "Completed"
  | "Cancelled";

export interface Customer {
  id: number;
  full_name: string;
}

export interface Product {
  id: number;
  name: string;
}

export interface OrderItem {
  id: number;
  product_id: number;
  product: Product;
  quantity: number;
  price: number;
  total: number;
}

export interface Order {
  id: number;
  customer_id: number;
  customer: Customer;
  payment_method: string;
  status: OrderStatus;
  subtotal: number;
  tax: number;
 grand_total: number;
  created_at: string;
  items: OrderItem[];
}

File: backend/app/schemas/order.py
from datetime import datetime

from pydantic import BaseModel
from pydantic import ConfigDict


class OrderItemCreate(BaseModel):
    product_id: int
    quantity: int


class OrderCreate(BaseModel):
    customer_id: int
    payment_method: str
    items: list[OrderItemCreate]


class OrderItemResponse(BaseModel):
    id: int
    product_id: int
    quantity: int
    price: float
    total: float

    model_config = ConfigDict(
        from_attributes=True
    )


class OrderResponse(BaseModel):
    id: int

    customer_id: int

    payment_method: str

    status: str

    subtotal: float

    tax: float

    grand_total: float

    created_at: datetime

    items: list[OrderItemResponse]

    model_config = ConfigDict(
        from_attributes=True
    )